planner/type_join.go

```diff
@@ -556,6 +558,17 @@ func (join *invertibleTypeJoin) processSecondResult(secondDocs []core.Doc) (any,
 }
 
 func (join *invertibleTypeJoin) Next() (bool, error) {
```

Comment:
thought: This function is really difficult to understand.

Reply:
Yes, and IMO the variable names don't really pair up well either, and they actually represent different things in different situations.

Reply:
https://github.com/sourcenetwork/defradb/issues/2577